Add GameManager struct

kesyog opened the pull request:
LFG

kesyog committed:
Add GameManager struct

mlodato517 commented on src/lib.rs:
```diff
@@ -16,15 +16,45 @@ impl Manager {
+    fn start_up(&self) {}
 
-    fn shut_down() {}
+    fn shut_down(&self) {}
```
For the future, these could (should?) be `&mut self.

mlodato517 commented on src/lib.rs:
```diff
@@ -16,15 +16,45 @@ impl Manager {
+struct GameManager {
+    manager: Manager,
+    game_over: bool,
+    frame_time: usize,
```
For the future, this could be a `Duration` so the "millisecond-ness" is more explicit.

mlodato517 reviewed (approved):
Definitely seems reasonable to me for now, especially because we're still not sure how this will be used :shipit:!

kesyog committed:
Switch Manager to be a trait

* Switch Manager to be a trait rather than a struct. I'm making an
  educated guess that it will be used polymorphically and this seems
  more Rust-y.
* Address PR feedback, using Duration for time and fixing mutability for
  Manager's methods

kesyog commented:
@mlodato517 took your suggestions and also went ahead and switched `Manager` to a trait. `Manager` doesn't serve much purpose as a helper struct, and it seems like it'll be more useful as an interface.
